doc/src/manual/interfaces.md

```diff
@@ -5,6 +5,81 @@ A lot of the power and extensibility in Julia comes from a collection of informa
 receive those functionalities, but they are also able to be used in other methods that are written
 to generically build upon those behaviors.
 
+## [Instance Properties](@id man-instance-properties)
+
+### [Overloading `object.field`](@id overloading-object-field)
+
+| Methods to implement              | Default definition           | Brief description                                                                     |
+|:--------------------------------- |:---------------------------- |:------------------------------------------------------------------------------------- |
+| `propertynames(x::ObjType, private::Bool=false)` | `fieldnames(typeof((x))`     | Returns a tuple of the properties (`x.property`) of an object `x`. `private=true` returns fieldnames intended to be kept as private |
+| `getproperty(x::ObjType, s::Symbol)`       | `getfield(x, s)`     | Returns property `s` of `x`. `x.s` calls `getproperty`.  |
+| `setproperty!(x::ObjType, s::Symbol, v)`   | `setfield!(x, s, v)` | Sets property `s` of `x` to `v`. `x.s = v` calls `setproperty!`. |
+
+Sometimes, it is desirable to change how end-user interacts with the fields of an object. 
+Instead of granting direct access to type fields, an extra layer of abstraction between 
+the user and the code can be provided overloading `object.field`. Properties are what the 
+user *sees of* the object, fields what the object *actually is*. 
```

Comment:
(Note that `setproperty!` should return `v`.)

Reply:
(This seems to be missing from the docstrings of `setfield!` and `setproperty!`)

Reply:
Yep, it is not said there 😞